core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -627,6 +631,26 @@ public void run() {
         }).drain();
   }
 
+  @Override
+  public void networkAvailable() {
+    channelExecutor.executeLater(
+        new Runnable() {
+          @Override
+          public void run() {
+            if (!nameResolverStarted || shutdown.get()) {
```

Comment:
nameResolverStarted should be reset to false when nameResolver is shutdown.

You are skipping the subchannels for the case `!nameResolverStarted` because we know that the channel is in idle mode and all subchannels are shut down. Because it won't harm to call the subchannels anyway, I would prefer not relying on that implication, which is not obvious at first glance. This would be easier to reason about:

```java
if (shutdown.get()) {
  return;
}
if (nameResolverStarted) {
  nameResolver.refresh();
}
for (...
...
```

Reply:
Done